app/controllers/UserProfileController.java

```diff
@@ -59,4 +70,20 @@ public Result updateUserProfile() throws Exception {
         
         return okResult("Profile updated.");
     }
+    
+    public Result createExternalIdentifier() throws Exception {
+        UserSession session = getAuthenticatedSession();
+
+        ExternalIdentifier externalId = ExternalIdentifier.fromJson(requestToJSON(request()));
```

Comment:
You can use BaseController.parseJson(request, clazz). parseJson wraps both the requestToJSON() call and the Jackson call, so you can leverage Jackson directly instead of building a custom fromJson() method.